include/seqan3/alignment/pairwise/alignment_algorithm.hpp

```diff
@@ -0,0 +1,170 @@
+    constexpr alignment_algorithm & operator=(alignment_algorithm &&)      = default;
+    ~alignment_algorithm()                                                 = default;
+
+    /*!\brief Constructs the algorithm with the passed configuration.
+     * \param cfg The configuration to be passed to the algorithm.
+     *
+     * \details
+     *
+     * The configuration is copied once to the heap during construction and maintained by a std::shared_ptr.
+     * The configuration is not passed to the function-call-operator of this function object, in order to avoid
+     * incompatible configurations between the passed configuration and the one used during configuration of this
+     * class. Further will the function object be stored in a std::function which requires copyable objects and
+     * in parallel executions the function object must be copied as well.
+     */
+    explicit constexpr alignment_algorithm(config_t const & cfg) : cfg_ptr{new config_t(cfg)}
+    {}
+    //!}
+
+    /*!\brief Invokes the actual alignment computation given two sequences.
+     * \tparam    first_batch_t  The type of the first sequence (or packed sequences); must model std::ForwardRange.
+     * \tparam    second_batch_t The type of the second sequence (or packed sequences); must model std::ForwardRange.
+     * \param[in] first_batch    The first sequence (or packed sequences).
+     * \param[in] second_batch   The second sequence (or packed sequences).
+     */
+    template <std::ranges::ForwardRange first_batch_t, std::ranges::ForwardRange second_batch_t>
+    auto operator()(first_batch_t const & first_batch, second_batch_t const & second_batch)
+    {
+        assert(cfg_ptr != nullptr);
+
+        // We need to allocate the score_matrix and maybe the trace_matrix.
+        this->allocate_score_matrix(first_batch, second_batch);
+
+        // Initialize cache variables to keep frequently used variables close to the CPU registers.
+        auto cache = this->setup_cache(get<align_cfg::gap>(*cfg_ptr).value);
+        // auto cache = this->setup_cache(gap_scheme{gap_score{-1}, gap_open_score{-10}});
+
+        initialize(cache);
+
+        compute(first_batch, second_batch, cache);
+
+        using result_t = typename select_result_type<first_batch_t, second_batch_t, config_t>::type;
+        result_t res{};
+
+        // Choose what needs to be computed.
+        if constexpr (config_t::template exists<align_cfg::result<detail::with_score_type>>())
+        {
+            auto last_col = this->active_column();
+            get<1>(res) = get<0>(*(std::ranges::end(last_col) - 1));
+            return res;
+        }
+        return res;
+
+    }
+
+protected:
+
+    /*!\brief Initializes the first column of the dynamic programming matrix.
+     * \tparam         cache_t The cache type.
+     * \param[in,out]  cache   The cache holding hot variables.
+     */
+    template <typename cache_t>
+    void initialize(cache_t & cache)
```

Comment:
`initialize` sounds like a constructor of alignment_algorithm. 
Consider a more descriptive name, e.g. `fill_1st_column`.

Reply:
Mhmm, I am not against it per sé, but don't really see any improvements. The text books call this step initialization and the function is documented and no user will ever touch it.

Reply:
I will though make it private instead of protected.